### PR TITLE
Removed redundant greatest_common_divisor code

### PR DESCRIPTION
**Pull request by Siddikpatel**
### Describe your change:

Previously all the files that needed greatest_common_divisor (aka gcd), used to define the method instead of just importing from Maths directory's greatest_common_divisor.py file. I removed thos definitions and imported gcd method from Maths folder. Fixes #8098

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by Siddikpatel** [comment]
It says organize imports. Don't know what to do about that. Otherwise there are no issues

**Comment by tianyizheng02** [comment]
@Siddikpatel Please run `ruff --fix .` locally to have ruff auto-sort the imports

**Comment by Siddikpatel** [comment]
Fixed the import issues!

File: maths/greatest_common_divisor.py [hunk 1]
```diff
@@ -1,37 +1,26 @@
 """
 Greatest Common Divisor.
 
-Wikipedia reference: https://en.wikipedia.org/wiki/Greatest_common_divisor
+Wikipedia reference: https://en.wikipedia.org/wiki/greatest_common_divisor
```
Comment (tianyizheng02) [inline maths/greatest_common_divisor.py:4]:
```suggestion
Wikipedia reference: https://en.wikipedia.org/wiki/Greatest_common_divisor
```
The original URL is correct: it's capitalized

File: maths/greatest_common_divisor.py [hunk 2]
```diff
@@ -1,37 +1,26 @@
...
+    4
+    >>> greatest_common_divisor(4, 7)
+    1
+    >>> greatest_common_divisor(0, 10)
+    10
     """
-    return abs(b) if a == 0 else greatest_common_divisor(b % a, a)
+    while b:
+        a, b = b, a % b
+    return a
```
Comment (tianyizheng02) [inline maths/greatest_common_divisor.py:23]: Please don't remove this recursive implementation. The iterative version is already implemented by `gcd_by_iterative`. If the recursive version is too inefficient to work in other files, then rename the two functions.

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

**Comment by tianyizheng02** [comment]
I just remembered that the Python `math` module already has a `gcd` function. @cclauss, do you think it's better for our files to simply import gcd from math? Or is it better for our files to use our own implementation?

File: maths/greatest_common_divisor.py (hunk 2 above)
Comment (Siddikpatel) [reply maths/greatest_common_divisor.py]: Oki. Fixed it.

**Comment by tianyizheng02** [comment]
BTW, @Siddikpatel, there are even more duplicate implementations of gcd if you search for "gcd" in the repo (for example, in `maths/carmichael_number.py`), so those could be removed as well

**Comment by Siddikpatel** [comment]
> BTW, @Siddikpatel, there are even more duplicate implementations of gcd if you search for "gcd" in the repo (for example, in `maths/carmichael_number.py`), so those could be removed as well

and, what about those files that imports gcd from math module of python?

**Review by cclauss — APPROVED** [review]

**Comment by tianyizheng02** [comment]
> and, what about those files that imports gcd from math module of python?

Those are fine—it's fine to import functions as helper functions as long as it doesn't include the core of the actual algorithm

**Review by tianyizheng02 — APPROVED** [review]
Looks good, but the merge conflict will need to be fixed. I can't fix it myself because I don't have write permission, so could you fix it?

**Comment by Siddikpatel** [comment]
> Looks good, but the merge conflict will need to be fixed. I can't fix it myself because I don't have write permission, so could you fix it?

Ok I am confused now.. It says merge conflicts in `modular_division.py`.

**Comment by tianyizheng02** [comment]
OK, I see what happened: a previous PR moved `blockchain/modular__division.py` to the `maths` directory, and that's causing a merge conflict here because you still have the file in the `blockchain` directory. I'm guessing that the conflict will be resolved if you move the file to the `maths` directory as well?

**Comment by Siddikpatel** [comment]
Yup, it is fixed now.

**Comment by tianyizheng02** [comment]
It says the file was deleted... did you not move it into the maths module?

**Comment by Siddikpatel** [comment]
> It says the file was deleted... did you not move it into the maths module?

I've moved it to maths directory.
Screenshot (160)

**Comment by quant12345** [comment]
@tianyizheng02 @cclauss A couple of days ago `carmichael_number` - add doctests. Do I need to delete this pull request (in 'Files changed' there is `gcd`)? At the bottom there was an inscription about a merge conflict. I made edits taking into account greatest_common_divisor. Now there is no warning. But will there be problems if there is a merger?

**Comment by tianyizheng02** [comment]
No, there shouldn't be any problems. This PR has already been merged so it can't be "deleted" anyway. For your other PR, GitHub will merge it into the updated codebase, the one with this PR already merged.

**Pull request merged**